### ISSUE TITLE
Modal - allow simple object scope

### ISSUE THREAD

**Issue by amichair** [angular-ui/bootstrap#3147]
It seems that currently when opening a Modal, the scope option value must be an actual angular scope object instance. This requires a bunch of boilerplate code to create/inject/pass/destroy (depending on the calling code structure, services etc.), and is quite cumbersome.

It would be great if a simple js object (which is not already an angular scope instance) could also be passed as a scope, just like with regular views, and the modal code will automatically turn it into a proper scope instance (and also call $destroy when the modal is closed).

**Comment by chandrewz** [comment]
I believe I am running into this problem too.

When I go with <datepicker></datepicker> in my modal, this is the only time datepicker appears. As soon as I attempt to use it with an input, there is no datepicker. I believe that is related to the scope.

@amichair are you able to get datepicker to appear in the modal?

**Comment by tdhsmith** [comment]
:+1:

Admittedly, I don't ask out of necessity, but because my preferred syntactic sugar is using isolated scopes to pass in template configurations. E.g. something like:

```javascript
scope: {
    title: {
        text: "Content not saved!",
        class: "alert"
    },
    buttons: [{text: "OK"}],
    ...
}
```
Creating a scope instance ahead of time is messy, and using a controller just to set variables is kind of semantically silly.

I imagine others might have some heavier use cases with full-fledged binding, but yeah, mine is just selfish desires about syntax. :wink:

**Comment by tdhsmith** [comment]
For reference, I'm now emulating the above by using `extend` and an anonymous scope:
```javascript
var modalInstance = $modal.open({
	templateUrl: 'standardModal',
	scope: angular.extend($scope.$new(true), {
					buttons: [{text: "OK"}],
					title: {text: "Registration successful!"},
					body: {text: "You have registered with the department [...]"}
				})
});
```

**Comment by amichair** [comment]
@chandrewz - I haven't tried a datepicker. Are you sure your issue is related to this one? It's hard to tell from your comment.

@tdhsmith from the [angular scope docs](https://docs.angularjs.org/guide/scope#scope-life-cycle) it seems like the created scope needs to be $destroyed by its creator, otherwise you'll have a memory leak and might increase your $digest time by having it continue to process the orphan scopes. I'm not sure in your case if you have to keep a reference to the anonymous scope or the extended object, but something needs to be destroyed when the modal is closed.

For the time being, I just use something like
```javascript
$scope.close = function() { modalInstance.close(); $scope.$destroy(); };
```
which is called from the view, but there are probably nicer ways of doing it.

Hopefully this issue will be closed and these workarounds will become moot :-)

**Comment by tdhsmith** [comment]
@amichair True, true! I realized that a bit later, but I'm glad you clarified a bit. I'm still new to managing object lifetimes manually.

I'm hedging that question for the moment because my modals are notices about being redirected off the page, so I can kind of cheat away the issue of memory leaks, but if I come up with a succinct rewrite I'll post it here. :smile:

**Comment by wesleycho** [comment]
I don't think we are interested in supporting this as originally stated.

A better option would be to create a controller for the modal, and take advantage of `controllerAs`  & `bindToController`, which is currently supported in `master` and will be released this weekend as 0.13.2.
